lib/processor/ceva/cpu.h

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2017, Pinecone Inc. and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * @file	cpu.h
+ * @brief	CPU specific primatives
+ */
+
+#ifndef __METAL_CEVA_CPU__H__
+#define __METAL_CEVA_CPU__H__
+
+#define metal_cpu_yield()
+
+#ifdef TEAKLITE4
+#define __sync_synchronize()
```

Comment:
Is this function used to flush cache? could you explain what is this function is used for?

Reply:
Since the compiler for CEVA' TEAKLITE4 series DSP doesn't implement __sync_synchronize builtin function, I have to define a dummy here to avoid the linker error.

Reply:
just want to know if the compiler doesn't implement this function, is the application trying to use it?
If yes, what is this function do? Usually who define this function?

Reply:
Yes, function in lib/compiler/gcc/atomic.h(e.g. atomic_store, atomic_load, atomic_thread_fence...) call __sync_synchronize.
Normally, atomic function provide by toolchain vendor and put into libgcc.a.
But, since CEVA TL42x DSP doesn't support the out of order execution and the mulitcore architecture, __sync_synchronize doesn't have any meaning on TL42x and CEVA forget to provide a dummy implementation here to improve the compatibility.

Reply:
As your DSP doesn't support out of order execution and multicore, looks like it is atomic still work with this dummy implementation. Looks fine for me, just better to have some comments about this definition.

Reply:
Ok, I will add some comment here.